pkg/authentication/firebase.go

```diff
@@ -44,3 +45,26 @@ func NewFirebaseWithTokenVerifier(firebaseAuth FirebaseTokenVerifier) Authentica
 		firebaseAuth: firebaseAuth,
 	}
 }
+
+// firebaseRefresher uses the firebase application to refresh the keys used to verify the token signature.
+// The firebase.App has an internal mechanism to avoid repeating this operation for every request.
+type firebaseRefresher struct {
+	app *firebase.App
+}
+
+// VerifyIDToken gets a new public key in case a key rotation has been requested, and verifies the given token.
+func (auth *firebaseRefresher) VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error) {
+	client, err := auth.app.Auth(ctx)
```

Comment:
Does it make sense to generate this client and store it inside `firebaseRefresher` when calling `NewFirebaseRefresher` or is anything refreshed when `Auth` is called?

Reply:
Thanks for the question. `Auth` performs a single RPC call when public keys need to be refresh.

According to the internal Go SDK doc: **These keys get cached up to 24 hours, and therefore the RPC overhead gets amortized over many invocations of this function.**

Reply:
I'm looking at the firebase code and `Auth(ctx)` creates and returns a new client every time it's called. It does cache keys and refreshes them as needed when calling methods on the returned client, which I believe is what you were suggesting.

I suggest that we add a `*auth.Client` property to `firebaseRefresher` and call `auth.app.Auth(ctx)` during initialization in `NewFirebaseRefresher`. The Firebase client will handle the refreshes automatically when calling `VerifyIDToken`.

Because this will remove the need to initialize the client in this method, we can also remove this method altogether, as it will be identical to just calling Firebase's `VerifyIDToken` (which is a good thing! Less code for us to maintain).